packages/core-utils/src/route.js

```diff
@@ -1,31 +1,151 @@
   if (!route.id) return null;
-  const operatorId = route.id.split(":")[0].toLowerCase();
-  return getTransitOperatorFromId(operatorId, transitOperators);
+  const feedId = route.id.split(":")[0];
+  let agencyId;
+  if (route.agency) {
+    // This is returned in the OTP Route model
+    agencyId = route.agency.id;
+  } else if (route.agencyId) {
+    // This is returned in the OTP RouteShort model (such as in the IBI fork)
+    agencyId = route.agencyId;
+  } else {
+    return null;
+  }
+  return getTransitOperatorFromFeedIdAndAgencyId(
+    feedId,
+    agencyId,
+    transitOperators
+  );
 }
 
+// The functions below are for enhanced route sorting functions for the route
+// viewer on OTP-react-redux.
+// They address route ordering issues discussed in
+// https://github.com/opentripplanner/otp-react-redux/pull/123 and
+// https://github.com/opentripplanner/otp-react-redux/pull/124.
+
 /**
- * The functions below are for enhanced route sorting functions for
- * the route viewer on OTP-react-redux.
- * They address route ordering issues discussed in
- * https://github.com/opentripplanner/otp-react-redux/pull/123 and
- * https://github.com/opentripplanner/otp-react-redux/pull/124.
+ * A large comparator value that can safely be used in mathematical sort
+ * comparisons to place things at the ned of lists
  */
+const END_OF_LIST_COMPARATOR_VALUE = 999999999999;
+
+/**
+ * Returns a transit operator comparator value given a route and an optional
+ * transitOperators config value. This function will do its best to handle all
+ * kinds of input data as certain deployments of an implementing webapp may have
+ * incomplete data and certain versions of OTP might not have a modified
+ * implementation of the RouteShort model.
+ *
+ * @param  {object} route Either an OTP Route or RouteShort model
+ * @param  {array} transitOperators transitOperators from config
+ * @return {mixed} this could return a string value (the route's agency name) if
+ *   the transitOperators value is not defined. Otherwise an integer will be
+ *   returned.
+ */
+function getTransitOperatorComparatorValue(route, transitOperators) {
+  // if the transitOperators is undefined, use the route's agency name as the
+  // comparator value
+  if (!transitOperators) {
+    // OTP Route
+    if (route.agency) return route.agency.name;
+    // OTP RouteShort (base OTP repo or IBI fork)
+    if (route.agencyName) return route.agencyName;
+    // shouldn't happen as agency names will be defined
+    return "zzz";
+  }
+
+  // find operator associated with route
+  const transitOperator = getTransitOperatorFromOtpRoute(
+    route,
+    transitOperators
+  );
+
+  // if transit operator not found, return infinity
+  if (!transitOperator) return END_OF_LIST_COMPARATOR_VALUE;
+
+  // return the transit operator's sort value or END_OF_LIST_COMPARATOR_VALUE if
+  // the sort value is not defined
+  return typeof transitOperator.order !== "undefined"
+    ? transitOperator.order
```

Comment:
Do we care if the order field type is a number (vs. string)? Probably not, but just flagging.

Reply:
Changed the check from being undefined to being a number in 5163a219c2ab5072d9338041435293fda8877c42